src/clj/game/core/hosting.clj

```diff
@@ -23,11 +23,21 @@
   [card target]
   (when (and card target) (or (same-card? card target) (has-ancestor? (:host card) target))))
 
+(defn handle-card-is-uninstalled
+  "If a card is hosted (uninstalled) from being installed and active, then call it's `leave-play` fn"
+  [state side card {:keys [installed] :as target}]
+  (when-let [leave-play-fn (and (active? (get-card state target))
+                                (not installed)
+                                (installed? (get-card state target))
+                                (:leave-play (card-def target)))]
```

Comment:
move this to the front of the `and` cuz it's both worth checking otherwise without such an fn

Reply:
hmm, I'm not quite sure what you mean.

Reply:
i mean that because `and` early exits, we should check that a :leave-play fn exists before we check if the card is active or installed or anything else. many cards will be active but only those with :leave-play fns matter.

Reply:
oh I see, yeah that's fair. I was just trying to be cute with the `when-let` heh